rcore/qemu.c

```diff
@@ -0,0 +1,96 @@
+#include "qemu.h"
+#include "task.h"
+#include "log.h"
+#include "rebbleos.h"
+
+extern const QemuEndpoint qemu_endpoints[];
+
+#define STACK_SZ_QEMU configMINIMAL_STACK_SIZE + 600
+
+static TaskHandle_t _qemu_task;
+static StackType_t _qemu_task_stack[STACK_SZ_QEMU];
+static StaticTask_t _qemu_task_buf;
+static uint8_t _qemu_packet_buffer[QEMU_MAX_DATA_LEN];
+
+static void _qemu_thread(void *pvParameters);
+static void _qemu_handle_packet();
+
+uint8_t qemu_init(void)
+{
+    hw_qemu_init();
+    _qemu_task = xTaskCreateStatic(_qemu_thread,
+                                   "QEMU", STACK_SZ_QEMU, NULL,
+                                   tskIDLE_PRIORITY + 3UL,
+                                   _qemu_task_stack, &_qemu_task_buf);
+    return INIT_RESP_OK;
+}
+
+static QemuEndpointHandler _qemu_find_endpoint_handler(uint16_t protocol)
+{
+    const QemuEndpoint *endpoint = qemu_endpoints;
+    while (endpoint->handler != NULL)
+    {
+        if (endpoint->protocol == protocol)
```

Comment:
If this is not true do we end up looping forever? What circumstance could cause this?